File: mario.c
#include <cs50.h>
#include <stdio.h>

int getHeight();
void drawPyramid();
int checkHeight();

int main(void)
{
    int height = getHeight();
    height = checkHeight(height);
    drawPyramid(height);
}


int getHeight()
{
    printf("Height: ");
    int i = get_int(); 
    return i;
}

void drawPyramid(int height)
{
    
    for (int layer = 0; layer < height; layer++)
    {
        int blocks = (1 + layer);
        
        for (int n = 0; n < (height - 1 - layer); n++)
            {
                printf(" ");
            }
        for (int n = blocks; n > 0; n--)
            {
                printf("#");
            }
        printf("  ");
        for (int n = blocks; n > 0; n--)
            {
                printf("#");
            }
        
        printf("\n");
    }
}
    


int checkHeight(int height)
{
    if (height < 0 || height > 23)
    {
        do
        {
            printf("Height: ");
            height = get_int();
        }
        while (height < 0 || height > 23);
        {
            return height;
        }
        
    }
    else if (0 == height && height < 24)
    {
        return height;
    }
    return height; 
}

File: mario.py
import cs50

#define main
def main():
    height = getHeight()
    checkHeight(height)
    drawPyramid(height)

#get the height requested
def getHeight():
    print("Height: ", end="")
    return cs50.get_int()
    
#check the height value
def checkHeight(height):
    if height < 0 or height > 23:
        while True:
            print ("Height: ", end="")
            height = cs50.get_int()
            if height < 0 or height > 23:
                break
        return height
    elif height >= 0 and height < 24:
        return height
    return height
    
#take the height to draw pyramid    
def drawPyramid(height):
    layer = 0
    for layer in range(height):
        blocks = 1 + layer
        for n in range(height - 1 - layer):
            print(" ", end="")
        for o in range(blocks, 0, -1):
            print("#", end="")
        print ("  ", end="")
        for p in range(blocks, 0, -1):
            print("#", end="")
        print("")
        layer += 1
        
#call main
if __name__ == "__main__":
    main()